### PR TITLE
[release-v2.6] Infracloud - Fix rancher-version and kube-version annotations

### PR DESCRIPTION
**Pull request by PennyScissors**
Issues:
https://github.com/rancher/rancher/issues/34125

This PR modifies the rancher-version annotation to improve consistency (and also [this](https://github.com/rancher/rancher/blob/4199cac04da5174351d99255695cce9b3b555073/pkg/image/charts.go#L205-L211)), and the kube-version annotation to be a closed range as part of an effort to improve the validation process when a new Kubernetes minor version is released. This will help automating the process, and generating a Kubernetes support matrix for all charts in the future (Not there yet, but moving in that direction). Also modifies the rancher-version annotation to improve consistency.

Note all charts where it is not clear if they support K8s 1.22, 1.23 and 1.24 (We don't have a support matrix available) have a kube-version cut off at `< 1.22.0-0` since it is the most common case and should've been set when 1.22 was released. 

**Please review these constraints and leave a comment if any of these constraint need to be updated to add support for 1.22, 1.23, and/or 1.24.**

### CODE REVIEW

**Comment by PennyScissors** [comment]
@samkulkarni20 Can you help review this? I need confirmation the ranges for the kube-version annotations I updated in this PR as correct. In some cases the charts had a kubeVersion already which I just matched in the kube-version annotation, but in others I had to use the default and I want to make sure any of the charts where I put < 1.22 actually supports 1.22 and over and needs to be updated.

**Review by samkulkarni20 — APPROVED** [review]
All annotations, including the kube-version are correct. Approving the PR.
For gatekeeper support matrix, I've started a discussion [here](https://github.com/orgs/open-policy-agent/discussions/213)

File: charts/rancher-gatekeeper/100.0.0+up3.5.1/Chart.yaml [hunk 1]
```diff
@@ -2,10 +2,11 @@ annotations:
   catalog.cattle.io/auto-install: rancher-gatekeeper-crd=match
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: OPA Gatekeeper
+  catalog.cattle.io/kube-version: '>= 1.16.0-0 < 1.22.0-0'
   catalog.cattle.io/namespace: cattle-gatekeeper-system
   catalog.cattle.io/os: linux
   catalog.cattle.io/provides-gvr: config.gatekeeper.sh.config/v1alpha1
-  catalog.cattle.io/rancher-version: < 2.6.1-0
```
Comment (aiyengar2) [inline charts/rancher-gatekeeper/100.0.0+up3.5.1/Chart.yaml:8]: This seems like an intentional change to hide the chart after 2.6.1 right? We should look into why this was necessary, but based on this we should be releasing:

```
'>= 2.6.0-0 < 2.6.1-0'
```

File: packages/rancher-gatekeeper/generated-changes/patch/Chart.yaml.patch [hunk 2]
```diff
@@ -5,11 +5,11 @@
 +  catalog.cattle.io/auto-install: rancher-gatekeeper-crd=match
 +  catalog.cattle.io/certified: rancher
 +  catalog.cattle.io/display-name: OPA Gatekeeper
-+  catalog.cattle.io/kube-version: '>= 1.16.0-0'
++  catalog.cattle.io/kube-version: '>= 1.16.0-0 < 1.22.0-0'
```
Comment (aiyengar2) [inline packages/rancher-gatekeeper/generated-changes/patch/Chart.yaml.patch:8]: let's just mark this chart doNotRelease in release-v2.6 and not modify the `packages/` changes. Same for istio.

**Review by aiyengar2 — CHANGES_REQUESTED** [review]

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
Can you also double check the concern from https://github.com/rancher/charts/pull/1928#pullrequestreview-1017839314? We should make sure releasing a chart does not lessen the kubeVersion we expect it to be supported on on release

File: charts/rancher-gatekeeper/100.0.0+up3.5.1/Chart.yaml (hunk 1 above)
Comment (PennyScissors) [reply charts/rancher-gatekeeper/100.0.0+up3.5.1/Chart.yaml:8]: Good catch, definitely missed that < in this one!

**Comment by PennyScissors** [comment]
@aiyengar2 Update gatekeeper to match Rancher's k8s support matrix. As for Istio I verified it against our matrix, but the Istio upstream versions support slightly older k8s versions so what this PR has seems correct based on this table https://istio.io/latest/docs/releases/supported-releases/#support-status-of-istio-releases

File: charts/rancher-gatekeeper/100.0.1+up3.6.0/Chart.yaml [hunk 3]
```diff
@@ -2,9 +2,11 @@ annotations:
   catalog.cattle.io/auto-install: rancher-gatekeeper-crd=match
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: OPA Gatekeeper
+  catalog.cattle.io/kube-version: '>= 1.16.0-0 < 1.24.0-0'
```
Comment (aiyengar2) [inline charts/rancher-gatekeeper/100.0.1+up3.6.0/Chart.yaml:5]: This was released in 2.6.1 https://github.com/rancher/rancher/issues/34589

According to https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-6-1, this should have only been tested up to k8s 1.21; are we sure 1.22 and 1.23 are supported?

File: charts/rancher-gatekeeper/100.1.0+up3.7.1/Chart.yaml [hunk 4]
```diff
@@ -2,11 +2,11 @@ annotations:
   catalog.cattle.io/auto-install: rancher-gatekeeper-crd=match
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: OPA Gatekeeper
-  catalog.cattle.io/kube-version: '>= 1.16.0-0'
+  catalog.cattle.io/kube-version: '>= 1.16.0-0 < 1.25.0-0'
```
Comment (aiyengar2) [inline charts/rancher-gatekeeper/100.1.0+up3.7.1/Chart.yaml:5]: This was released in Rancher 2.6, which supports up to 1.23

https://github.com/rancher/rancher/issues/37376

https://www.suse.com/assets/EN-2.6.5SupportMatrix-300422-0116-26.pdf

Are we sure this works on 1.24?

**Review by aiyengar2 — CHANGES_REQUESTED** [review]

**Review by aiyengar2 — APPROVED** [review]
LGTM

**Comment by PennyScissors** [comment]
Closed in favor of this PR to consolidate PRs from all teams: https://github.com/rancher/charts/pull/1964

**Pull request closed without merging**